ch2driver/pytpcc/util/rand.py

```diff
@@ -30,135 +30,151 @@
 # -----------------------------------------------------------------------
 
 import random
+import numpy as np
+import constants
 import string
 from . import nurand
 
 SYLLABLES = [ "BAR", "OUGHT", "ABLE", "PRI", "PRES", "ESE", "ANTI", "CALLY", "ATION", "EING" ]
 
-nurandVar = None # NURand
-def setNURand(nu):
-    global nurandVar
-    nurandVar = nu
-## DEF
-
-def NURand(a, x, y):
-    """A non-uniform random number, as defined by TPC-C 2.1.6. (page 20)."""
-    global nurandVar
-    assert x <= y
-#    assert nurand != None
-    if nurandVar is None:
-        setNURand(nurand.makeForLoad())
-    if a == 255:
-        c = nurandVar.cLast
-    elif a == 1023:
-        c = nurandVar.cId
-    elif a == 8191:
-        c = nurandVar.orderLineItemId
-    else:
-        raise Exception("a = " + a + " is not a supported value")
+class Rand:
+
+    def __init__(self, datagenSeed=None):
+        self.nurandVar = None # NURand
+        self.rng = random.Random()
+        self.nprng = np.random.default_rng()
+        if datagenSeed != constants.CH2_DATAGEN_SEED_NOT_SET:
+            self.rng.seed(datagenSeed)
+            self.nprng = np.random.default_rng(datagenSeed)
+
+    def setNURand(self, nu):
+        self.nurandVar = nu
+    ## DEF
+
+    def NURand(self, a, x, y):
```

Comment:
nitpick: given the refactoring going on this file, we could rename this to be consistent with method/function naming conventions. Right now it looks like a class. We could rename it `nurand` or `nuRand`.

Reply:
Done, will do a follow up merge.